src/ngx_pagespeed.cc

```diff
@@ -544,7 +544,8 @@ ngx_int_t copy_response_headers_to_ngx(
     }
 
     if (STR_EQ_LITERAL(name, "Cache-Control")) {
-      ps_set_cache_control(r, const_cast<char*>(value_gs.c_str()));
+      ps_set_cache_control(r, string_piece_to_pool_string(
+          r->pool, value_gs.c_str()));
```

Comment:
Another idea: what would you think of changing the `gs` suffix to `unsafe` or something to indicate that it's not a pool string?